src/server/transaction.cc

```diff
@@ -1194,6 +1200,32 @@ bool Transaction::NotifySuspended(TxId committed_txid, ShardId sid) {
   return false;
 }
 
+void Transaction::LogJournalOnShard(EngineShard* shard) {
+  // TODO: For now, we ignore non shard coordination.
+  if (shard == nullptr)
+    return;
+
+  if ((cid_->opt_mask() & CO::WRITE) == 0)
+    return;
+
+  auto journal = shard->journal();
+  if (journal == nullptr)
+    return;
+
+  // TODO: Handle complex commands like LMPOP correctly once they are implemented.
+  journal::Entry::Payload entry_payload;
+  if (unique_shard_cnt_ == 1) {
+    CHECK(!full_args_.empty());
+    entry_payload = full_args_;
+  } else if (args_.empty()) {
+    entry_payload = make_pair(facade::ToSV(full_args_.front()), ArgSlice{});
```

Comment:
can we write 
if (unique_shard_cnt_ == 1 ||  args_.empty()) 
entry_payload = full_args_;
else 
entry_payload =
        make_pair(facade::ToSV(full_args_.front()), ShardArgsInShard(shard->shard_id()));

?

Reply:
I would have to change ShardArgInShard to return an empty slice. Currently there is a `DCHECK(!args_.empty())` in ShardArgInShard.

I think it would be safe to change the mechanics, but it would no longer guard the other transaction code from bugs (incorrect shard access)

Reply:
I saw the check inside ShardArgInShard and I think its better to keep it there for now.
My suggestion was not to move the args_.empty() case to the initialisation using ShardArgsInShard but to the one using full_args_

Reply:
Indeed